[Feat] 스터디 생성 페이지 추가

dmstmdrbs opened the pull request:
공통으로 사용 될 수 있는 스타일을 정리했어요!
  - CommonStyles.ts에서 확인할 수 있어요
  - Flex박스들과 각 페이지에 적용할 수 있는 PageContainer를 만들었어요
 
스터디 생성 페이지를 추가 했어요!
  - /create 경로에서 확인할 수 있어요
  - 스터디 시간 계획을 위한 컴포넌트를 분리했어요 (CreateTimePlan.tsx)
 
공통으로 사용 할 Button 컴포넌트를 추가했어요
  - 아직은 기본적인 옵션만 적용할 수 있어요

dmstmdrbs committed:
feat: add UI logic for Create page

스터디 생성 UI 추가
  - 시간 설정을 위한 CreateTimePlan 컴포넌트 추가
	- 요일 버튼 DayButton 컴포넌트 추가
	- 요일 추가 시 시간 설정 필요
ESLint rules 수정
  - operator-linebreak off
공통 컴포넌트 추가
  - Input 컴포넌트
	- TextArea 컴포넌트

dmstmdrbs committed:
Merge branch 'feat/router' of https://github.com/gimsamgyo/summer_study_planner into feat/createStudy

dmstmdrbs committed:
style: modify layout width & height

레이아웃 구조 변경
	- 최상위 div 삭제

dmstmdrbs committed:
feat: add Button component

공통 Button 컴포넌트 추가 및 적용

dmstmdrbs committed:
Merge branch 'main' into feat/createStudy

Coreight98 commented on src/components/CreateStudy/CreateTimePlan.tsx:
```diff
@@ -0,0 +1,113 @@
+  });
+
+  const onClickCancel = () => {
+    onCancel();
```
파라미터가 들어가지 않는 함수는 별도로 감쌀 필요는 없는거 같아요!

dmstmdrbs replied:
추후에 컴포넌트 내부에서의 로직이 들어갈 여지가 있지 않을까하여 따로 감쌌는데 제가 보기에도 현재는 필요 없을 것 같습니다..! 지금은 재명님 피드백에 맞춰 수정하고 추후에 로직 추가하게 되면 그때 다시 감싸는 방향으로 수정해보겠습니다

Coreight98 commented on src/components/CreateStudy/CreateTimePlan.tsx:
```diff
@@ -0,0 +1,113 @@
+      <FlexCol>
+        <ItemsCenter>
+          <Label>시작 시간</Label>
+          <input
```
제가 잘 본건지 모르겠습니다만, 시작 시간이 끝시간보다 늦을때에 대한 에러 핸들링이 없는거 같아요!
저번에 사용해주신 react-hook-form 사용해주시면 에러 핸들링에 용이하실거 같다고 생각되요! 아니면 yup만 사용하셔서 유효성 검사만 하셔도 좋을거 같다고 생각해요!

dmstmdrbs replied:
맞게 파악하셨습니다! 아직 에러 핸들링에 대한 부분은 없습니다.
react-hook-form 도입해서 수정해보겠습니다!

Coreight98 commented on src/components/Day/DayButton.tsx:
```diff
@@ -0,0 +1,30 @@
+  isFilled,
+}: {
+  content: string;
+  onClick: () => void;
```
단순히 클릭한다는 동작 함수 네이밍 보다는 이 버튼의 동작에 대한 네이밍으로 바꾸면 좋을거 같다고 생각해요!
현재 이 버튼을 day에 관련된 버튼으로 만들어서, 날짜를 체크하는 동작에 관한 함수라고 생각되는데, isCheck와 같은 네이밍이 좋을거 같아요!

dmstmdrbs replied:
isChecked와 같은 이름은 체크 되었는지에 대한 여부인 것 같아서 setCheck나 setDay, setDayChecked 와 같은 이름은 어떨까요?

Coreight98 commented on src/components/Input.tsx:
```diff
@@ -0,0 +1,19 @@
+    outline: 1px solid coral;
+  }
+`;
+const Input = ({ ...rest }) => (
```
지금은 어떤 입력창 컴포넌트가 모두 정의되어 있지 않아서 ...rest 사용하셨지만, 나중에는 유지 보수 해야할 레거시라고 생각해요!
나중을 생각하기 보다는 현재 사용된 기능들에 초점을 두고 개발하는게 더 바람직하다고 생각해서(입력창이 다양하게 나오지 않을거라는 기대감? ㅎㅎ) 나중에 수정하면 좋을거 같아요!!

dmstmdrbs replied:
딱 용도에 맞게 그때그때 추가하라는 말씀이시죠??
말씀하신대로 현재 사용되는 기능들에 맞춰서 수정해보겠습니다!

Coreight98 commented on src/components/Layout.tsx:
```diff
@@ -39,14 +44,14 @@ const Layout = ({ title, canGoBack, children }: LayoutProps) => {
         {canGoBack && <BackButton onClick={onClickGoBack}>&larr;</BackButton>}
         {title && <span>{title}</span>}
       </Header>
       <Children>{children}</Children>
```
나중에 hoc로 분류해서 폴더관리해도 좋을거 같아요!

Coreight98 commented on src/pages/Create.tsx:
```diff
@@ -1,11 +1,152 @@
+  time: { startTime: string; endTime: string };
+};
+const Create = () => {
+  const [plannedDays, setPlannedDay] = useState<PlannedDayType[]>([] as PlannedDayType[]);
```
set 함수에도 days를 붙여주면 통일성이 있을거 같아요!|
plannedDays 보다 좋은 네이밍이 있을거 같은데, 현재로서는 떠오르지 않네용.. ㅎㅎ

dmstmdrbs replied:
s가 빠졌었군요..!

Coreight98 commented on src/pages/Create.tsx:
```diff
@@ -1,11 +1,152 @@
+      const currentDayIndex = currentPlannedDays.findIndex(
+        (plannedDay) => plannedDay.day === isDayClicked,
+      );
+      if (currentDayIndex === -1) {
```
현재 같은 날짜가 선택되었을 때 처리하는 함수가 맞을까요?
맞다면, yub이나, form에 대한 유효성 검사를 한다면 없앨 수 있는 로직같아요!

dmstmdrbs replied:
이미 시간 계획이 저장 된 날인지 판별 후에 

1. 시간 계획이 없는 날이면 time plan을 새로 추가한다.
2. 시간 계획이 저장 된 날이면 기존의 time plan을 가져온 후 새로운 데이터로 수정한다.

if/else 분기에 따라 위와 같이 동작하는 함수입니다!

Coreight98 replied:
아 같은 날이여도 시간대가 다를 수 있어서 그렇게 처리하신거죠??

Coreight98 commented on src/components/Button.tsx:
```diff
@@ -0,0 +1,30 @@
+import styled from 'styled-components';
+
+interface IButtonStyle {
+  primary?: boolean;
```
primary에 대한 bool 처리 보다는 primary | filled로 타입을 주는 건 어떤가요? 확장성을 고려할때 다른 버튼 타입도 나올 수 있다고 생각해서요!

dmstmdrbs replied:
primary와 filled를 다른 속성으로 보라는 말씀이시겠죠?
다른 속성으로 보는게 여러 상황에 대비하기에 좋을 것 같네요 ㅎㅎ

dmstmdrbs committed:
refactor: delete wrapper function

파라미터가 없는 함수를 감싸는 wrapper 함수 삭제

dmstmdrbs committed:
refactor: rename function of DayButton's onClick prop

동작과 관련된 이름으로 props의 onClick 이름 변경

dmstmdrbs committed:
refactor: rename setter function of plannedDays

복수형 이름을 위한 s 추가

dmstmdrbs committed:
Merge branch 'feat/createStudy' of https://github.com/gimsamgyo/summer_study_planner into feat/createStudy

dmstmdrbs committed:
style: add width property in PageContainer

모바일 화면을 위한 width 100% 추가

dmstmdrbs committed:
chore: add dependency, react-hook-form

dmstmdrbs committed:
feat: add form validation using reacct-hook-form & modify input, textarea components

react-hook-form을 이용한 폼 유효성 검사 추가
react-hook-form 추가에 따른 Input, TextArea 컴포넌트 리팩토링
  - Label,ErrorMessage 추가
	- Label 추가에 따른 레이아웃 변경
	- 이외 Props 정리

dmstmdrbs committed:
chore: disable rule for object-curly-newline
